common/src/test/java/com/hartwig/serve/common/classification/matchers/NegativeMatcherTest.java

```diff
@@ -0,0 +1,27 @@
+package com.hartwig.serve.common.classification.matchers;
+
+import com.google.common.collect.Sets;
+import org.junit.Test;
+
+import java.util.Set;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class NegativeMatcherTest {
+
+    private static final Set<String> NEGATIVE_EVENTS = Sets.newHashSet("negative");
+    private static final Set<String> NEGATIVE_BLACKLIST_EVENTS = Sets.newHashSet("HRD", "MSI");
+
+    @Test
+    public void canAssessWhetherEventIsGeneLevelEvent() {
+        NegativeMatcher matcher = new NegativeMatcher(NEGATIVE_EVENTS,
+                NEGATIVE_BLACKLIST_EVENTS);
+
+        assertTrue(matcher.matches("AKT1", "AKT1 negative"));
+        assertTrue(matcher.matches("TP53", "TP53 negative"));
+
+        assertFalse(matcher.matches("HRD", "HRD negative"));
+        assertFalse(matcher.matches("MSI", "MSI negative"));
```

Comment:
The last condition at the end of NegativeMatcher is never reached by any of the tests I think (event equals gene), so could maybe add a couple of assertions for expected behaviour for that as well. 

I'm not really clear on why the gene doesn't matter, e.g. matches("AKT1", "TP53 negative") would match as true I think. But I guess that can never come up or is otherwise by design?

Reply:
Ok, thanks for the clarification.